src/main/java/uk/gov/digital/ho/hocs/casework/domain/repository/JsonConfigFolderReader.java

```diff
@@ -0,0 +1,67 @@
+package uk.gov.digital.ho.hocs.casework.domain.repository;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.stereotype.Service;
+import uk.gov.digital.ho.hocs.casework.application.LogEvent;
+import uk.gov.digital.ho.hocs.casework.domain.exception.ApplicationExceptions;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URISyntaxException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+@Service
+public abstract class JsonConfigFolderReader {
+
+    private final ObjectMapper objectMapper;
+
+    protected JsonConfigFolderReader(ObjectMapper objectMapper) {
+        this.objectMapper = objectMapper;
+    }
+
+    protected <T extends CaseTypeObject<V>, V> Map<String, V> readValueFromFolder(TypeReference<T> reference) {
+        var folder = Thread.currentThread().getContextClassLoader().getResource(String.format("config/%s", getFolderName()));
```

Comment:
There is a `getResources` method to this that returns an Enumerable and likely could be used. While this code is likely less performant, we'd either have to do weird `while...do` logic or convert to a `EnumerationSpliterator`. 

I've chosen readability over simplicity/performance here.